site/app/views/forum/ForumThreadView.php

```diff
@@ -997,97 +947,56 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
-
-            $return .= $this->core->getOutput()->renderTwigTemplate("forum/CategoriesForm.twig", [
-                "categories" => $categories,
-                "category_colors" => $category_colors
-            ]);
+        if(!$this->forumAccess()){
+            $this->core->redirect($this->core->buildUrl(array('component' => 'navigation')));
+            return;
         }
 
-		$return .= <<<HTML
-		<div class="content forum_content">
-		
-HTML;
-
-	$buttons = array(
-		array(
-			"required_rank" => 4,
-			"display_text" => 'Back to Threads',
-			"style" => 'position:relative;top:3px;float:right;',
-			"link" => array(true, $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread'))),
-			"optional_class" => '',
-			"title" => 'Back to threads',
-			"onclick" => array(false)
-		)
-	);
-
-	$thread_exists = $this->core->getQueries()->threadExists();
+        $this->core->getOutput()->addBreadcrumb("Discussion Forum", $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread')));
+        $this->core->getOutput()->addBreadcrumb("Create Thread", $this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread')));
 
-	$return .= $this->core->getOutput()->renderTwigTemplate("forum/ForumBar.twig", [
-								"forum_bar_buttons_right" => $buttons,
-								"forum_bar_buttons_left" => [],
-								"show_threads" => false,
-								"thread_exists" => $thread_exists,
-								"show_more" => false
-	]);
+        $this->core->getOutput()->addInternalJs('drag-and-drop.js');
+        $this->core->getOutput()->addVendorJs('flatpickr/flatpickr.js');
+        $this->core->getOutput()->addVendorJs('jquery.are-you-sure/jquery.are-you-sure.js');
 
+        $this->core->getOutput()->addInternalCss('flatpickr.min.css');
 
-		$return .= <<<HTML
+        $categories = "";
+        $category_colors;
 
-			<h3 style="margin-bottom:10px;"> Create Thread </h3>
-			<form style="margin-right: 15px; margin-left:15px;" id="thread_form" method="POST" action="{$this->core->buildUrl(array('component' => 'forum', 'page' => 'publish_thread'))}" enctype="multipart/form-data">
-HTML;
-                $this->core->getOutput()->addInternalJs("flatpickr.js");
-                $this->core->getOutput()->addInternalCss('flatpickr.min.css');
-				$return .= $this->core->getOutput()->renderTwigTemplate("forum/ThreadPostForm.twig", [
-					"show_title" => true,
-					"show_post" => true,
-					"post_textarea_large" => true,
-                    "show_lock_date" => true,
-					"post_content_placeholder" => "Enter your post here...",
-					"show_categories" => true,
-					"post_box_id" => 1,
-					"attachment_script" => true,
-					"show_anon" => true,
-					"show_thread_status" => true,
-					"show_announcement" => true,
-					"show_editcat" => true,
-					"submit_label" => "Submit Post",
-				]);
-			$return .= <<<HTML
-			</form>
-		</div>
-HTML;
+        if($this->core->getUser()->getGroup() <= 3){
```

Comment:
Please use `$this->core->getUser()->accessGrading()`

Reply:
Done!